### PR TITLE
Edge onclick

### PR DESCRIPTION
**Pull request by IwanHoogenboom**
Made edge onclick.

### CODE REVIEW

File: src/main/java/programminglife/gui/controller/GuiController.java [hunk 1]
```diff
@@ -443,6 +444,16 @@ private void mouseClick(double x, double y, boolean shiftPressed) {
                 }
                 graphController.highlightClicked(segment, shiftPressed);
             }
+            else if (clickedOn instanceof DrawableEdge) {
+                DrawableEdge edge = (DrawableEdge) clickedOn;
+                System.out.println("JEmoeder");
```
Comment (YDutchy) [inline src/main/java/programminglife/gui/controller/GuiController.java]: Remove this......

File: src/main/java/programminglife/gui/controller/GraphController.java [hunk 2]
```diff
@@ -431,4 +418,8 @@ public void highlightClicked(DrawableSegment segment, boolean shiftPressed) {
     public void resetZoom() {
         this.subGraph.setZoomLevel(1);
     }
+
+    public SubGraph getSubGraph() {
+        return subGraph;
+    }
```
Comment (MeAmAnUsername) [inline src/main/java/programminglife/gui/controller/GraphController.java]: Why did you add this?

File: src/main/java/programminglife/model/drawing/SubGraph.java [hunk 3]
```diff
@@ -313,6 +313,112 @@ public void setZoomLevel(double zoomLevel) {
...
 
     /**
+     * On click method
+     * @param x location
+     * @param y location
+     * @return Drawable on which clicked. if nothing null.
+     */
+    public Drawable onClick(double x, double y) {
+        Drawable clicked = null;
+        System.out.println(x + " "+ y);
```
Comment (MeAmAnUsername) [inline src/main/java/programminglife/model/drawing/SubGraph.java]: remove this

File: src/main/java/programminglife/model/drawing/SubGraph.java [hunk 4]
```diff
@@ -313,6 +313,112 @@ public void setZoomLevel(double zoomLevel) {
...
+        //TODO implement this with a tree instead of iterating.
+        if (this == null) {
+            return null;
+        }
```
Comment (MeAmAnUsername) [inline src/main/java/programminglife/model/drawing/SubGraph.java]: `this == null` is impossible

File: src/main/java/programminglife/model/drawing/SubGraph.java [hunk 5]
```diff
@@ -313,6 +313,112 @@ public void setZoomLevel(double zoomLevel) {
...
+
+        for (DrawableNode drawableNode : this.getNodes().values()) {
+            if (x >= drawableNode.getLocation().getX() && y >= drawableNode.getLocation().getY()
+                    && x <= drawableNode.getLocation().getX() + drawableNode.getWidth()
+                    && y <= drawableNode.getLocation().getY() + drawableNode.getHeight()) {
+                clicked = drawableNode;
+                return clicked;
```
Comment (MeAmAnUsername) [inline src/main/java/programminglife/model/drawing/SubGraph.java]: Can be replaced with `return drawableNode`

File: src/main/java/programminglife/model/drawing/SubGraph.java [hunk 6]
```diff
@@ -313,6 +313,112 @@ public void setZoomLevel(double zoomLevel) {
     }
 
     /**
+     * On click method
+     * @param x location
+     * @param y location
+     * @return Drawable on which clicked. if nothing null.
+     */
+    public Drawable onClick(double x, double y) {
+        Drawable clicked = null;
```
Comment (MeAmAnUsername) [inline src/main/java/programminglife/model/drawing/SubGraph.java]: Can be left out

File: src/main/java/programminglife/model/drawing/SubGraph.java [hunk 7]
```diff
@@ -313,6 +313,112 @@ public void setZoomLevel(double zoomLevel) {
...
+                    && x <= drawableNode.getLocation().getX() + drawableNode.getWidth()
+                    && y <= drawableNode.getLocation().getY() + drawableNode.getHeight()) {
+                clicked = drawableNode;
+                return clicked;
+            }
+        }
+
+        clicked = onClickEdge(x, y);
+
+        return clicked;
```
Comment (MeAmAnUsername) [inline src/main/java/programminglife/model/drawing/SubGraph.java]: Can be replaced with `return onClickEdge(x, y);`

File: src/main/java/programminglife/model/drawing/SubGraph.java [hunk 8]
```diff
@@ -313,6 +313,112 @@ public void setZoomLevel(double zoomLevel) {
...
+    }
+
+    /**
+     * check if clicked on an edge.
+     *
+     * @param x locatie
+     * @param y locatie
```
Comment (MeAmAnUsername) [inline src/main/java/programminglife/model/drawing/SubGraph.java]: Dutch

File: src/main/java/programminglife/model/drawing/SubGraph.java [hunk 9]
```diff
@@ -313,6 +313,112 @@ public void setZoomLevel(double zoomLevel) {
...
+            for (DrawableNode right : this.getChildren(left)) {
+                    if (calculateEdge(left, right, x, y)) {
+
+                       return new DrawableEdge(left, right);
+                    }
+
+                }
+            }
+
+        System.out.println("=====================");
```
Comment (MeAmAnUsername) [inline src/main/java/programminglife/model/drawing/SubGraph.java]: remove this

File: src/main/java/programminglife/model/drawing/SubGraph.java [hunk 10]
```diff
@@ -305,6 +305,112 @@ public void setZoomLevel(double zoomLevel) {
...
```
Comment (YDutchy) [inline src/main/java/programminglife/model/drawing/SubGraph.java]: Also remove this

File: src/main/java/programminglife/model/drawing/SubGraph.java [hunk 11]
```diff
@@ -313,6 +313,112 @@ public void setZoomLevel(double zoomLevel) {
...
+        Layer leftLayer = layers.get(leftLayerIndex);
+
+        for (DrawableNode left : leftLayer.getNodes()) {
+            for (DrawableNode right : this.getChildren(left)) {
+                    if (calculateEdge(left, right, x, y)) {
+
+                       return new DrawableEdge(left, right);
+                    }
+
+                }
```
Comment (MeAmAnUsername) [inline src/main/java/programminglife/model/drawing/SubGraph.java]: This has way too many newlines
